lib/CallBackery/qooxdoo/callbackery/source/class/callbackery/ui/plugin/Action.js

```diff
@@ -246,18 +243,65 @@ qx.Class.define("callbackery.ui.plugin.Action", {
             },this);
         },
         _addUploadButton: function(cfg,btCfg,getFormData){
-            var form = new uploadwidget.UploadForm('uploadFrm','upload');
-            form.setParameter('name',cfg.name);
-            form.setParameter('key',btCfg.key);
-            form.setLayout(new qx.ui.layout.HBox());
-            this.add(form);
             var file;
             if (btCfg.btnClass == 'toolbar') {
-                file = new uploadwidget.UploadToolbarButton('file', this.xtr(btCfg.label));
+                file = new callbackery.ui.form.UploadToolbarButton(this.xtr(btCfg.label));
             }
             else {
-                file = new uploadwidget.UploadButton('file', this.xtr(btCfg.label));
+                file = new callbackery.ui.form.UploadButton(this.xtr(btCfg.label));
             }
+
+            let that = this;
+            file.addListener('changeFileName', function(e){
+                let formData = new FormData();
+                formData.append("file", e.getData());
+                formData.append("name", "onboardingBulkUpload");
+                formData.append("key", "upload_bulk_invite");
+
+                callbackery.data.Server.getInstance().callAsyncSmart(function(cookie){
+                    formData.append('xsc',cookie);
+
+                    var xhr = new XMLHttpRequest();
```

Comment:
please use qooxdoo implementation http://www.qooxdoo.org/apps/apiviewer/#qx.io.request.Xhr

Reply:
I don't immediately get this to work. I don't know how to make the upload file appear in the correct location in the request.